python-package/lightgbm/dask.py

```diff
@@ -1174,7 +1174,7 @@ def fit(  # type: ignore[override]
         eval_init_score: Optional[List[_DaskCollection]] = None,
         eval_metric: Optional[_LGBM_ScikitEvalMetricType] = None,
         **kwargs: Any
-    ) -> "DaskLGBMClassifier":
+    ) -> "_DaskLGBMModel":
```

Comment:
I noticed you've removed these Dask changes from this PR and they aren't included in the other mypy-related PR you opened (#5731).

Are you interested in trying them out? If not, I'll fix these issues related to `lightgbm.dask` return types.

Reply:
Since we're doing `# type: ignore[override]` it seemed kind of redundant to me.

Reply:
It is not. No worries, I'll attempt this particlar fix.

Reply:
Created #5756.

To add more details on why the presence of `# type: ignore[override]` doesn't make this concern redundant...that check and these errors about return types are about different things.

The `[override]` check is saying "hey, you told me this was a child of `{ParentClass}`, but it has a method with a different signature than `{ParentClass}`, which means it can't be automatically used anywhere that `{ParentClass}` is used".

The errors about return types are saying "hey you told me that this `fit()` method returns an instance of `{ChildClass}` but it's returning an instance of `{ParentClass}`". That's a different thing, and important to fix for at least 2 reasons:

1. `{ChildClass}` might define additional attributes that `{ParentClass}` doesn't (e.g. extra configuration captured through the constructor)
2. People writing code that depends on `lightgbm` may wish to perform checks like `isinstance(model, lgb.DaskLGBMRegressor)`. Such checks would fail today if called on the result of `fit()`, since it's returning an internal class `lgb._DaskLGBMModel`.